core/state_processor_rip7560.go

```diff
@@ -321,36 +348,40 @@ func ApplyRip7560ExecutionPhase(config *params.ChainConfig, vpr *ValidationPhase
 	if err != nil {
 		return nil, err
 	}
-	root := statedb.IntermediateRoot(true).Bytes()
 	var paymasterPostOpResult *ExecutionResult
 	if len(vpr.PaymasterContext) != 0 {
 		paymasterPostOpResult, err = applyPaymasterPostOpFrame(vpr, executionResult, evm, gp, statedb, header)
-		root = statedb.IntermediateRoot(true).Bytes()
 	}
 	if err != nil {
 		return nil, err
 	}
 
-	cumulativeGasUsed :=
+	gasUsed :=
 		vpr.ValidationUsedGas +
 			vpr.DeploymentUsedGas +
 			vpr.PmValidationUsedGas +
 			executionResult.UsedGas
 	if paymasterPostOpResult != nil {
-		cumulativeGasUsed +=
+		gasUsed +=
 			paymasterPostOpResult.UsedGas
 	}
 
-	receipt := &types.Receipt{Type: vpr.Tx.Type(), PostState: root, CumulativeGasUsed: cumulativeGasUsed}
-
-	// Set the receipt logs and create the bloom filter.
-	receipt.Logs = statedb.GetLogs(vpr.Tx.Hash(), header.Number.Uint64(), header.Hash())
+	receipt := &types.Receipt{Type: vpr.Tx.Type(), TxHash: vpr.Tx.Hash(), GasUsed: gasUsed, CumulativeGasUsed: gasUsed}
 
 	if executionResult.Failed() || (paymasterPostOpResult != nil && paymasterPostOpResult.Failed()) {
 		receipt.Status = types.ReceiptStatusFailed
 	} else {
 		receipt.Status = types.ReceiptStatusSuccessful
 	}
+
+	refundPayer(vpr, statedb, gasUsed)
+
+	// Set the receipt logs and create the bloom filter.
+	blockNumber := header.Number
+	receipt.Logs = statedb.GetLogs(vpr.TxHash, blockNumber.Uint64(), common.Hash{})
```

Comment:
Why pass `common.Hash{}` instead of `header.Hash()` ? I am sure something will break, won't it?

Reply:
I used to do it. but this value is meaningless, as it calculates a hash over a header that will get modified.
the DerivedFields() method will fill in those values. better see zeros for missing fields rather than bogus value that might "look" correct, but aren't.